File: modul4/C/PRAK405_2310817210017_MUHAMMADNURHUDABAIHAQI.c
#include <stdio.h>

void hitungKelipatan(int n, int kelipatan) {
    int total = 0;

    for (int i = 1; i <= n; i++) {
        int baristotal = 0;
        printf("(");
        for (int h = i; h > 0; h--) {
            baristotal += h * kelipatan;
            printf("%d * %d", h, kelipatan);
            if (h > 1) {
                printf(" + ");
            }
        }
        printf(") = %d\n", baristotal);
        total += baristotal;
    }

    printf("%d\n", total);
}

int main() {
    int n, kelipatan;
    while(1){
    printf("");
    scanf("%d %d", &n, &kelipatan);
    if (n == 0) {
        break;
    }
    hitungKelipatan(n, kelipatan);
    printf("\n");
    }
    return 0;
}

File: modul4/Python/PRAK405_2310817210017_MUHAMMADNURHUDABAIHAQI.py
def hitung_kelipatan(n, kelipatan):
    total = 0

    for i in range(1, n + 1):
        baris_total = 0
        print("(", end="")
        for j in range(i, 0, -1):
            baris_total += j * kelipatan
            print(f"{j} * {kelipatan}", end="")
            if j > 1:
                print(" + ", end="")
        print(f") = {baris_total}")
        total += baris_total

    print(total)

if __name__ == "__main__":
    while True:
        input_str = input("")
        if input_str.lower() == "0":
            break
        n, kelipatan = map(int, input_str.split())
        hitung_kelipatan(n, kelipatan)
        print()
